riscv/processor.cc

```diff
@@ -397,6 +397,33 @@ void state_t::reset(processor_t* const proc, reg_t max_isa)
   const reg_t henvcfg_init = (proc->extension_enabled(EXT_SVPBMT) ? HENVCFG_PBMTE : 0);
   csrmap[CSR_HENVCFG] = henvcfg = std::make_shared<henvcfg_csr_t>(proc, CSR_HENVCFG, henvcfg_mask, henvcfg_init, menvcfg);
 
+  const reg_t mstateen0_mask = (proc->extension_enabled(EXT_ZFINX) ? MSTATEEN0_FCSR : 0) |
+                               MSTATEEN0_CS | MSTATEEN0_HENVCFG | MSTATEEN_HSTATEEN;
+  const reg_t mstateen0h_mask = MSTATEEN0H_HENVCFG | MSTATEENH_HSTATEEN;
+  const reg_t hstateen0_mask = (proc->extension_enabled(EXT_ZFINX) ? HSTATEEN0_FCSR : 0) |
+                               HSTATEEN0_CS | HSTATEEN0_SENVCFG | HSTATEEN_SSTATEEN;
+  const reg_t hstateen0h_mask = HSTATEEN0H_SENVCFG | HSTATEENH_SSTATEEN;
+  const reg_t sstateen0_mask = (proc->extension_enabled(EXT_ZFINX) ? SSTATEEN0_FCSR : 0) |
+                               SSTATEEN0_CS;
+  for (int i = 0; i < 4; i++) {
```

Comment:
Is Smstateen is not configured, none of these CSRs should exist.

Reply:
I think so.

Reply:
I see there is a check in `stateen_csr_t::verify_permissions()` that will trap in this case. But that dynamic check isn't necessary since Smstateen can't be dynamically disabled. Instead the check could be here in `processor_t::reset()` and skip the creation of the CSRs in the first place.

Reply:
OK. I'll change the check to reset later.